pkg/amqp-bunny/AmqpProducer.php

```diff
@@ -70,7 +101,7 @@ public function send(Destination $destination, Message $message): void
         InvalidMessageException::assertMessageInstanceOf($message, InteropAmqpMessage::class);
 
         try {
-            $this->doSend($destination, $message);
+            $this->doSendWithPossibleConfirm($destination, $message);
```

Comment:
The feature should support two behaviors:

* Sync mode. Publish a message and wait for confirmation.
* Async mode. Publish a message return the control and let a developer check the status whenever they want.

Reply:
How should we choose the behavior? Additional method, 2 separate modes or something set in the message itself?

If 2 modes - it's probably best to make them both available globally?

Reply:
Probably by attaching a property / header to `Message`. I doubt `Destination` has something that you could use (but I might be wrong).

Reply:
It could be a bool flag argument for `enableConfirmMode` that tells if the publishing sync or async (if async a developer should perform a check later by themselves)